src/js/components/Header/Header.js

```diff
@@ -1,19 +1,61 @@
-import React from 'react';
-
+import React, { useContext, useEffect } from 'react';
+import { ThemeContext } from 'styled-components';
+import { findScrollParent, useForwardedRef } from '../../utils';
 import { Box } from '../Box';
 
-const Header = React.forwardRef(({ ...rest }, ref) => (
-  <Box
-    align="center"
-    as="header"
-    direction="row"
-    flex={false}
-    justify="between"
-    gap="medium"
-    {...rest}
-    ref={ref}
-  />
-));
+const Header = React.forwardRef(({ sticky, ...rest }, ref) => {
+  const theme = useContext(ThemeContext);
+  const containerRef = useForwardedRef(ref);
+  useEffect(() => {
+    let scrollTop = 0;
+    const updateScrollDir = () => {
+      // target is its scroll parent
+      const target = findScrollParent(containerRef.current);
+      const container = containerRef.current;
```

Comment:
More just a variable naming suggestion. Would it make sense to rename:
`container` --> `header`
`containerRef` --> `headerRef`

?

Reply:
yeah that seems approprite

Reply:
done